Support multiple stacks

mattmb opened the pull request:
This makes it easier to create multiple stacks of the same app and environment. It adds a random string after the stack name so that it does not clash, this is associated with a tag, such as inactive or active.

This doesn't solve the whole problem because some resources still need renaming (i.e. dns-names, S3 buckets). However you no longer need to rename all references to an environment e.g. renaming prod to prod1.

A change is also needed to bootstrap-salt (PR to follow).

I _think_ this is backwards compatible. But it is more code than I wanted to write...

Needs some docs.

Commit:
Support multiple stacks

This makes it easier to create multiple stacks of the same app and
environment. It adds a random string after the stack name so that it
does not clash, this is associated with a tag, such as inactive or
active.

koikonom commented on bootstrap_cfn/fab_tasks.py:
```diff
@@ -148,16 +169,111 @@ def user(username):
+    cfn_config = get_config()
+    try:
+        r53_conn = get_connection(R53)
+        zone_name = cfn_config.data['master_zone']
```
What if I don't use a master zone? Isn't elb:hosted_zone the same ?

mattmb replied:
Looks like I've covered that by catching the KeyError and returning the legacy_name?

Commit:
Update README/CHANGELOG

koikonom commented on bootstrap_cfn/fab_tasks.py:
```diff
@@ -148,16 +169,111 @@ def user(username):
+
+
 def get_stack_name():
     """
```
As discussed, this and generate_stack_name should be one method.

Commit:
Combine get_stack_name and generate_stack_name

These functions were nearly the same. So combine them and have an
argument to separate them.
